go-controller/pkg/factory/factory.go

```diff
@@ -704,10 +735,16 @@ func NewNodeWatchFactory(ovnClientset *util.OVNNodeClientset, nodeName string) (
 		wf.apbRouteFactory.K8s().V1().AdminPolicyBasedExternalRoutes().Informer()
 	}
 
-	// need to configure OVS interfaces for Pods on secondary networks in the DPU mode.
+	if util.IsRouteAdvertisementsEnabled() {
+		wf.raFactory = routeadvertisementsinformerfactory.NewSharedInformerFactory(ovnClientset.RouteAdvertisementsClient, resyncInterval)
+		// make sure shared informer is created for a factory, so on wf.raFactory.Start() it is initialized and caches are synced.
+		wf.raFactory.K8s().V1().RouteAdvertisements().Informer()
+	}
+
+	// need to configure OVS interfaces for Pods on secondary networks in the DPU mode
 	// need to know what is the primary network for a namespace on the CNI side, which
 	// needs the NAD factory whenever the UDN feature is used.
-	if config.OVNKubernetesFeature.EnableMultiNetwork || config.OVNKubernetesFeature.EnableNetworkSegmentation {
+	if config.OVNKubernetesFeature.EnableMultiNetwork && (config.OVNKubernetesFeature.EnableNetworkSegmentation || config.OvnKubeNode.Mode == types.NodeModeDPU) {
```

Comment:
Hmm which part? For `config.OVNKubernetesFeature.EnableMultiNetwork`, I explained on @pperiyasamy previous comment. For `config.OvnKubeNode.Mode == types.NodeModeDPU`, it was being checked before but I found on a rebase that it was removed so I added it back because it didn't seem correct to me: before network segmentation, we needed NADs on DPU. I don't how that has changed?